cirq/ops/common_gates.py

```diff
@@ -575,6 +578,28 @@ def __repr__(self) -> str:
             return 'CNOT'
         return 'CNOT**{!r}'.format(self.half_turns)
 
+    def on(self, *args: raw_types.QubitId,
+           **kwargs: raw_types.QubitId) -> gate_operation.GateOperation:
+        if kwargs and args:
+            raise ValueError('A mix of named and positional arguments has been '
+                             'specified for a CNOT operation.\n'
+                             'Be sure to choose either one of them')
+        for argkey in kwargs.keys():
```

Comment:
This loop and the following two conditions can be simplified into this:

```python
if set(kwargs.keys()) not in [{}, {'control', 'target'}]:
    raise ValueError("Expected 'control' and 'target' keyword arguments, or no keyword arguments, but got {}.".format(set(kwargs.keys()))
```

Or.. I guess the method could be written like so

```python
if not kwargs:
    return super().on(*args)
if not args and set(kwargs.keys) == {'control', 'target'}:
    return super().on(kwargs['control'], kwargs['target'])
raise ValueError("Expected two positional argument or else 'target' and 'control' keyword arguments. But got args={!r}, kwargs={!r}.".format(args, kwargs))
```

Reply:
I tried to find a good balance between readability and lines of code.
Thought the solution I came up with would be okey already, but I liked your suggestion better.